src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp

```diff
@@ -35,48 +36,41 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
     }
 
     std::map<std::string, std::vector<nebula::cpp2::ColumnDef>> edgeColumns;
-    for (auto const &element : properties.get_fields()) {
-        auto edgeName = element.first;
-        auto edgeType = getEdgeType(space, edgeName);
-        if (!edgeType.ok()) {
-            LOG(ERROR) << "Create Edge Index Failed: " << edgeName << " not exist";
-            resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
-            onFinished();
-            return;
-        }
+    auto edgeTypeRet = getEdgeType(space, edgeName);
+    if (!edgeTypeRet.ok()) {
+        LOG(ERROR) << "Create Edge Index Failed: " << edgeName << " not exist";
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
 
-        auto fieldsResult = getLatestEdgeFields(space, edgeType.value());
-        if (!fieldsResult.ok()) {
-            LOG(ERROR) << "Get Latest Edge Property Name Failed";
+    auto edgeType = edgeTypeRet.value();
+    auto fieldsResult = getLatestEdgeFields(space, edgeType);
+    if (!fieldsResult.ok()) {
+        LOG(ERROR) << "Get Latest Edge Property Name Failed";
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+    auto fields = fieldsResult.value();
+    std::vector<nebula::cpp2::ColumnDef> columns;
+    for (auto &field : fieldNames) {
+        auto iter = std::find_if(std::begin(fields), std::end(fields),
+                                 [field](const auto& pair) { return field == pair.first; });
```

Comment:
Do we have to make sure the field is the first N fields in schema? @bright-starry-sky

Reply:
Of course